### ISSUE TITLE
scan_csv with Categorical dtype -> PanicException

### ISSUE THREAD

**Issue by s-banach** [pola-rs/polars#5222]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

In polars==0.14.19, `scan_csv` with a Categorical dtype gives the error `"Cannot concat Categoricals coming from a different source. Consider setting a global StringCache."`  (In prior versions, we just get wrong results.)

(In general, I hope polars can support combining Categorical columns from two different dataframes.)

### Reproducible example

```python
import polars as pl

N = 5_000
pl.DataFrame({"x": ["A"] * N}).write_csv("test.csv")
df = pl.scan_csv("test.csv", dtypes={"x": pl.Categorical}).collect()
```

### Expected behavior

Don't crash.

### Installed versions

<details>

```
---Version info---
Polars: 0.14.19
Index type: UInt32
Platform: Windows-10-10.0.19044-SP0
Python: 3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:50:36) [MSC v.1929 64 bit (AMD64)]
---Optional dependencies---
pyarrow: 3.0.0
pandas: 1.5.0
numpy: 1.23.3
fsspec: <not installed>
connectorx: <not installed>
xlsx2csv: <not installed>
matplotlib: 3.6.1
```

</details>

**Comment by ritchie46** [comment]
Right.. we need to set the string cache ourselves.

In the meantime you can explicitly set the stringcache and your query should run.

**Comment by ritchie46** [comment]
fixed by #5225
